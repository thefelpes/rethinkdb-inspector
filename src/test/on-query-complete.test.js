// @flow
import inspect from '../';
import rethinkdbdash from 'rethinkdbdash';

it('should return timing information', async () => {
  const r = rethinkdbdash({
    pool: false,
  });
  const run = jest.fn(() => Promise.resolve('a'));
  r._Term.prototype.run = run;
  const onQueryComplete = jest.fn();
  inspect(r, {
    onQueryComplete,
  });
  await r
    .db('test')
    .table('bla')
    .get('asdf123')
    .run();
  expect(run).toHaveBeenCalledTimes(1);
  expect(onQueryComplete).toHaveBeenCalledTimes(1);
  expect(onQueryComplete.mock.calls[0][0]).toMatchSnapshot();
  expect(onQueryComplete.mock.calls[0][1].time).toBeGreaterThan(0);
});

it('should return size information', async () => {
  const r = rethinkdbdash({
    pool: false,
  });
  const run = jest.fn(() => Promise.resolve('a'));
  r._Term.prototype.run = run;
  const onQueryComplete = jest.fn();
  inspect(r, {
    onQueryComplete,
  });
  await r
    .db('test')
    .table('bla')
    .get('asdf123')
    .run();
  expect(run).toHaveBeenCalledTimes(1);
  expect(onQueryComplete).toHaveBeenCalledTimes(1);
  expect(onQueryComplete.mock.calls[0][0]).toMatchSnapshot();
  expect(onQueryComplete.mock.calls[0][1].size).toEqual(2);
});

it('should return size information for objects', async () => {
  const r = rethinkdbdash({
    pool: false,
  });
  const run = jest.fn(() => Promise.resolve({ test: true }));
  r._Term.prototype.run = run;
  const onQueryComplete = jest.fn();
  inspect(r, {
    onQueryComplete,
  });
  await r
    .db('test')
    .table('bla')
    .get('asdf123')
    .run();
  expect(run).toHaveBeenCalledTimes(1);
  expect(onQueryComplete).toHaveBeenCalledTimes(1);
  expect(onQueryComplete.mock.calls[0][0]).toMatchSnapshot();
  expect(onQueryComplete.mock.calls[0][1].size).toEqual(12);
});
