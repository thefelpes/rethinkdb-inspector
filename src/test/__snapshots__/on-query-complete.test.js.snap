// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return size information 1`] = `
"r.db(\\"test\\").table(\\"bla\\").get(\\"asdf123\\")
"
`;

exports[`should return size information for objects 1`] = `
"r.db(\\"test\\").table(\\"bla\\").get(\\"asdf123\\")
"
`;

exports[`should return timing information 1`] = `
"r.db(\\"test\\").table(\\"bla\\").get(\\"asdf123\\")
"
`;
